cpp/server/proxy.cc

```diff
@@ -0,0 +1,134 @@
+#include "server/proxy.h"
+
+#include <event2/http.h>
+#include <evhttp.h>
+
+#include "monitoring/monitoring.h"
+#include "net/url_fetcher.h"
+#include "proto/ct.pb.h"
+#include "server/json_output.h"
+#include "util/libevent_wrapper.h"
+
+
+namespace cert_trans {
+
+namespace {
+static Counter<std::string>* total_proxied_requests(
```

Comment:
Would add (at least) one empty line after the ```namespace```, but I would feel all right if you clumped the ```namespace```s together.